### PR TITLE
DBZ-1513 Centralize handling of non-data messages in SMT

### PR DESCRIPTION
**Pull request by jpechane**
Start for discussion on https://issues.jboss.org/browse/DBZ-1513?focusedCommentId=13791336&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13791336

### CODE REVIEW

File: debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public void canSkipMessagesWithoutDebeziumCdcEnvelopeDueToMissingSchemaName() {
 
         final SourceRecord eventRouted = router.apply(eventRecord);
 
-        assertThat(eventRouted).isNull();
+        assertThat(eventRouted).isSameAs(eventRecord);
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java]: The method name here (and the two below) should be adjusted accordingly.

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java [hunk 2]
```diff
@@ -198,7 +198,7 @@ public void shouldDropHeartbeatMessages() {
         // when
         SourceRecord transformed = transformation.apply(eventRecord);
 
-        assertThat(transformed).isNull();
+        assertThat(transformed).isSameAs(eventRecord);
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java]: Ah, so this was inconsistent then with the relational SMT then? +1 for aligning it, not sure why they ever were dropped here. Warrants a migration notes entry for sure.

File: debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java (hunk 1 above)
Comment (jpechane) [reply debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java]: @gunnarmorling Done

**Comment by gunnarmorling** [comment]
Thanks, @jpechane!

**Pull request merged**